src/Facets/RelayFacet.sol

```diff
@@ -0,0 +1,221 @@
+                )
+            )
+        );
+        address signer = ECDSA.recover(message, _relayData.signature);
+        if (signer != relaySolver) {
+            revert InvalidQuote();
+        }
+        _;
+    }
+
+    /// Constructor ///
+
+    constructor(address _relayReceiver, address _relaySolver) {
+        relayReceiver = _relayReceiver;
+        relaySolver = _relaySolver;
+    }
+
+    /// External Methods ///
+
+    /// @notice Bridges tokens via Relay
+    /// @param _bridgeData The core information needed for bridging
+    /// @param _relayData Data specific to Relay
+    function startBridgeTokensViaRelay(
+        ILiFi.BridgeData calldata _bridgeData,
+        RelayData calldata _relayData
+    )
+        external
+        payable
+        nonReentrant
+        onlyValidQuote(_bridgeData, _relayData)
+        refundExcessNative(payable(msg.sender))
+        validateBridgeData(_bridgeData)
+        doesNotContainSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
+    {
+        LibAsset.depositAsset(
+            _bridgeData.sendingAssetId,
+            _bridgeData.minAmount
+        );
+        _startBridge(_bridgeData, _relayData);
+    }
+
+    /// @notice Performs a swap before bridging via Relay
+    /// @param _bridgeData The core information needed for bridging
+    /// @param _swapData An array of swap related data for performing swaps before bridging
+    /// @param _relayData Data specific to Relay
+    function swapAndStartBridgeTokensViaRelay(
+        ILiFi.BridgeData memory _bridgeData,
+        LibSwap.SwapData[] calldata _swapData,
+        RelayData calldata _relayData
+    )
+        external
+        payable
+        nonReentrant
+        onlyValidQuote(_bridgeData, _relayData)
+        refundExcessNative(payable(msg.sender))
+        containsSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
+        validateBridgeData(_bridgeData)
+    {
+        _bridgeData.minAmount = _depositAndSwap(
+            _bridgeData.transactionId,
+            _bridgeData.minAmount,
+            _swapData,
+            payable(msg.sender)
+        );
+        _startBridge(_bridgeData, _relayData);
+    }
+
+    /// Internal Methods ///
+
+    /// @dev Contains the business logic for the bridge via Relay
+    /// @param _bridgeData The core information needed for bridging
+    /// @param _relayData Data specific to Relay
+    function _startBridge(
+        ILiFi.BridgeData memory _bridgeData,
+        RelayData calldata _relayData
+    ) internal {
+        // check if sendingAsset is native or ERC20
+        if (LibAsset.isNativeAsset(_bridgeData.sendingAssetId)) {
+            // Native
+
+            // Send Native to relayReceiver along with requestId as extra data
+            (bool success, bytes memory reason) = relayReceiver.call{
+                value: _bridgeData.minAmount
+            }(abi.encode(_relayData.requestId));
```

Comment:
I think we still need to enforce that this is a bytes32 when being called initially.

Reply:
I dont understand your response to be honest....where/how are we enforcing anything there? And why do we need to enforce?  Could you please elaborate?

Please consider leaving issues with open questions unresolved until the reviewer sees your response and can decide if the issue is in fact resolved during re-review

Reply:
The `requestId` will always be a `bytes32` so leaving the argument as `bytes32` ensures that noone can pass in a longer or shorter amount of bytes. The only reason we need to encode as bytes here is because solidity won't allow us to simply send `bytes32` as calldata in `call()`

Reply:
How would this be an issue if someone passed less/more bytes. Arent we validating them anyway (in the modifier)?

Just want to understand it and see if it actually wouldnt work without the manual encoding in the contract.